airflow/models/pool.py

```diff
@@ -161,12 +161,12 @@ def occupied_slots(self, session: Session):
         """
         from airflow.models.taskinstance import TaskInstance  # Avoid circular import
 
-        return (
+        return int((
             session.query(func.sum(TaskInstance.pool_slots))
             .filter(TaskInstance.pool == self.pool)
             .filter(TaskInstance.state.in_(list(EXECUTION_STATES)))
             .scalar()
-        ) or 0
+        ) or 0)
```

Comment:
`0` does not need to be converted, so I’d just cast the query part:

```python
return int(
    session.query(func.sum(TaskInstance.pool_slots))
    .filter(TaskInstance.pool == self.pool)
    .filter(TaskInstance.state.in_(list(EXECUTION_STATES)))
    .scalar()
) or 0
```

Reply:
@uranusjr I tried this first, but failed, when filter matched nothing, it returns `None` 😭

Reply:
Per [MySQL doc](https://dev.mysql.com/doc/refman/8.0/en/aggregate-functions.html#function_sum):
> If there are no matching rows, SUM() returns NULL.

Reply:
Oh… alright then.